catalyst/contrib/dl/callbacks/tiled_inference.py

```diff
@@ -0,0 +1,242 @@
+            predictions,
+            idx,
+            x_min,
+            y_min,
+            storage_size,
+            margins,
+            tile_size,
+        ) in zip(
+            output,
+            state.input["id"],
+            state.input["x"],
+            state.input["y"],
+            state.input["storage_size"],
+            state.input["margins"],
+            state.input["tile_size"],
+        ):
+            idx = idx.item()
+            x_min = x_min.item()
+            y_min = y_min.item()
+            storage_size = storage_size.cpu().numpy().tolist()
+            margins = margins.cpu().numpy().tolist()
+            tile_size = tile_size.cpu().numpy().tolist()
+
+            num_classes, *_ = predictions.shape
+            storage_size = Shape(*storage_size)
+            margins = Edges(*margins)
+            tile_size = Shape(*tile_size)
+
+            if idx not in self.allocated:
+                self.image_idx.append(idx)
+
+                storage = torch.zeros(
+                    (num_classes, storage_size.height, storage_size.width)
+                )
+                norm_mask = torch.zeros(
+                    (1, storage_size.height, storage_size.width)
+                )
+                weights = pyramid_weights(tile_size.height, tile_size.width)
+                self.allocated[idx] = Allocated(
+                    storage=storage, norm_mask=norm_mask, weights=weights,
+                )
+
+                self.margins[idx] = margins
+
+            x_max = x_min + tile_size.width
+            y_max = y_min + tile_size.height
+            crop_slice = (
+                slice(None),
+                slice(y_min, y_max),
+                slice(x_min, x_max),
+            )
+
+            storage = self.allocated[idx].storage.to(state.device)
+            norm_mask = self.allocated[idx].norm_mask.to(state.device)
+            weights = self.allocated[idx].weights.to(state.device)
+
+            storage[crop_slice] += predictions * weights
+            norm_mask[crop_slice] += weights
+
+            self.allocated[idx] = Allocated(
+                storage=storage, norm_mask=norm_mask, weights=weights,
+            )
+
+            if (norm_mask != 0.0).all().item():
+                self._store_ready(idx)
+                self.allocated.pop(idx)
+
+    def _store_ready(self, idx: int):
+        allocated = self.allocated[idx]
+        margins = self.margins[idx]
+        eps = torch.finfo(allocated.norm_mask.dtype).eps
+        norm_mask = torch.clamp_min(allocated.norm_mask, eps)
+        logits = allocated.storage / norm_mask
+        num_classes, h, w = logits.shape
+        crop_slice = (
+            slice(None),
+            slice(margins.top, h - margins.bottom),
+            slice(margins.left, w - margins.right),
+        )
+        logits = logits[crop_slice]
+
+        if num_classes == 1:
+            probs = torch.sigmoid(logits)
+        else:
+            probs = F.softmax(logits, dim=0)
```

Comment:
probably it is better to ask user to choose activation function
e.g. in satellite imaging one pixel can both describe weather (cloudless, ...) and type of ground (forests, ...)